symplyphysics/laws/quantities/quantity_is_molar_quantity_times_amount_of_substance.py

```diff
@@ -0,0 +1,49 @@
+from sympy import Eq, Symbol as SymSymbol
+from symplyphysics import (
+    units,
+    Quantity,
+    Symbol,
+    print_expression,
+    validate_input,
+)
+from symplyphysics.core.dimensions import assert_equivalent_dimension
+
+# Description
+## An extensive quantity of interest can be obtained by multiplying the corresponding molar-specific
+## intensive quantity by the amount of substance.
+
+# Law: X = X_m * n
+## X - extensive quantity
+## X_m - (intensive) molar quantity
+## n - amount of substance
+
+extensive_quantity = SymSymbol("extensive_quantity")
+molar_quantity = SymSymbol("molar_quantity")
+amount_of_substance = Symbol("amount_of_substance", units.amount_of_substance)
+
+law = Eq(extensive_quantity, molar_quantity * amount_of_substance)
+
+
+def print_law() -> str:
+    return print_expression(law)
+
+
+@validate_input(amount_of_substance_=amount_of_substance)
+def calculate_extensive_quantity(
+    molar_quantity_: Quantity,
+    amount_of_substance_: Quantity,
+) -> Quantity:
+    result_expr = law.rhs.subs({
+        molar_quantity: molar_quantity_,
+        amount_of_substance: amount_of_substance_,
+    })
+    result = Quantity(result_expr)
+
+    assert_equivalent_dimension(
+        molar_quantity_,
```

Comment:
Check result.

Reply:
![image](https://github.com/blackyblack/symplyphysics/assets/115647936/96727671-07b7-4db6-882a-94d56a669056)
The assertion function raises error messages about arguments, not returns. That's why I reformulated it in terms of argument, I didn't want to change `assert_equivalent_dimension`. If we change the error texts to be about the return value, it will be just alright.

Reply:
You can see that we use it for returned values as well: https://github.com/blackyblack/symplyphysics/blob/1d23238a7aa98f7410bd9e34b1d1e2f6ae2556d1/symplyphysics/core/quantity_decorator.py#L83
I agree that text is a bit misleading but it's ok.

Reply:
Hm, okay, now I see